packages/fxa-settings/src/components/App/index.tsx

```diff
@@ -105,137 +128,135 @@ export const App = ({
     }
   }, [metricsEnabled, config.sentry, config.version, loading, isSignedIn]);
 
-  useEffect(() => {
-    if (Storage.isLocalStorageEnabled(window)) {
-      let uniqueUserId = JSON.parse(
-        localStorage.getItem('__fxa_storage.currentAccountUid') || '{}'
-      );
+  return (
+    <Router basepath="/">
+      <AuthAndSetUpRoutes path="/*" />
+      <SettingsRoutes path="/settings/*" {...{ showRecoveryKeyV2 }} />
+    </Router>
+  );
+};
 
-      const accounts: Array<{ email: string; uid: string }> = JSON.parse(
-        localStorage.getItem('__fxa_storage.accounts') || '{}'
-      );
+const SettingsRoutes = ({
+  showRecoveryKeyV2,
+}: {
+  showRecoveryKeyV2?: boolean;
+} & RouteComponentProps) => {
+  // TODO: FXA-8098
+  // const { loading, error } = useInitialSettingsState();
+  const settingsContext = initializeSettingsContext();
 
-      const currentUser = Object.values(accounts).find(
-        (x) => x.uid === uniqueUserId
-      );
+  return (
+    <SettingsContext.Provider value={settingsContext}>
+      <ScrollToTop default>
+        <Settings path="/settings/*" {...{ showRecoveryKeyV2 }} />
+      </ScrollToTop>
+    </SettingsContext.Provider>
+  );
+};
 
-      if (currentUser) {
-        setEmail(currentUser.email);
-      }
-      setEmailLookupComplete(true);
-    }
-  }, [setEmail, email, setEmailLookupComplete]);
+const AuthAndSetUpRoutes = (_: RouteComponentProps) => {
```

Comment:
When trying to group these routes together, the best I could come up with was `NonSettings` or `AuthAndSetUp` as in, the routes mainly handle authentication and initial account set up. I went with `SetUp` since "account set up" would make it a verb - I actually called this Setup originally, lol. I'm totally open to another name; if/when we need a Context here we should name it similarly.

Reply:
Gotcha. IMO `AuthAndAccountSetupRoutes` makes is clearer, but what do I know. I never really thought about it but auth is a noun, so I'd say go setup to keep the parts of speech consistent. Or maybe `SignInSignUpRoutes`?

Reply:
I'm all about that clarity so I'll add `Account` at least, but I feel torn on `SetUp` vs `Setup`. 😂 Either way I think I like `AuthAndAccountSetupRoutes` or `AuthAndAccountSetUpRoutes` over `SignInSignUpRoutes` since there are other routes like pairing and an inline TOTP set up that don't quite fit into sign in/up.

Reply:
I'm not smart enough to comment on the English correctness of ["set up" vs "setup"](https://grammarist.com/spelling/set-up-vs-setup/), but a shameful <kbd>grep</kbd> tells me that `setup` is more common in our codebase:

```sh
git grep -Pi "SeTuP" -- ":(exclude)packages/123done" ":(exclude).yarn/" ":(exclude).github/" ":(exclude)docs/" ":(exclude).circleci/"
```

versus `setUp` which seems to be only used in fxa-content-server, and only in two test files:

```sh
git grep -P "\b[Ss]etUp" packages | grep -v "\.py:" | grep -Ev "setUp(date|load)"

packages/fxa-content-server/tests/functional/bounced_email.js:const setUpBouncedSignIn = thenify(function (email) {
packages/fxa-content-server/tests/functional/bounced_email.js:        .then(setUpBouncedSignIn())
packages/fxa-content-server/tests/functional/bounced_email.js:        .then(setUpBouncedSignIn(email))
packages/fxa-content-server/tests/functional/bounced_email.js:        .then(setUpBouncedSignIn())
packages/fxa-content-server/tests/functional/bounced_email.js:        .then(setUpBouncedSignIn())
packages/fxa-content-server/tests/server/metrics-unit.js:  var test = setUp();
packages/fxa-content-server/tests/server/metrics-unit.js:  var test = setUp();
packages/fxa-content-server/tests/server/metrics-unit.js:  var test = setUp();
packages/fxa-content-server/tests/server/metrics-unit.js:  var test = setUp(function () {
packages/fxa-content-server/tests/server/metrics-unit.js:  var test = setUp(function (headerName) {
packages/fxa-content-server/tests/server/metrics-unit.js:  var test = setUp(function (headerName) {
packages/fxa-content-server/tests/server/metrics-unit.js:  var test = setUp(function (headerName) {
packages/fxa-content-server/tests/server/metrics-unit.js:function setUp(requestGet) {
```

Reply:
I asked Valerie too and she thinks `Setup` is more correct. So I'll go with `AuthAndAccountSetupRoutes` 😎